### PR TITLE
Adding custom join messages.

### PR DESCRIPTION
**Pull request by zunda-arrow**
MCoding bot will say a custom join message when a user joins the server. Join Messages need to be disabled on the server before this is in production or two messages will send when someone joins.
**Needed Environment Variables**
WELCOME_CHANNEL=Channel ID

Custom join messages are stored in `assets/join_messages`. If the location is changed it also needs to be changed in `app/config.py`.  Theres only one rn so hopefully some people think of a few more.

### CODE REVIEW

File: app/bot.py [hunk 1]
```diff
@@ -175,6 +176,22 @@ async def on_message(self, message: discord.Message):
 
         await self.process_commands(message)
 
+    async def on_member_join(self,member: discord.Member):
```
Comment (Enderchief) [inline app/bot.py]: Change function to 
`async def on_member_join(self, member: discord.Member):` 
(notice space between `self,` and `member`

File: app/bot.py [hunk 2]
```diff
@@ -175,6 +176,22 @@ async def on_message(self, message: discord.Message):
...
+
+        channel_id = self.config.welcome_channel
+        if channel_id is None:
+            self.log("WELCOME_CHANNEL environment variable is not defined. Member join message will not be sent.")
+            return
+
+        channel = self.get_channel(channel_id)
+        join_message = random.choice(self.config.join_messages)
+        message = await channel.send(join_message.format(member=member.mention))
+        await message.add_reaction( "👋")
```
Comment (Enderchief) [inline app/bot.py]: remove extra space and change to `await message.add_reaction("👋")`

**Review by Enderchief — CHANGES_REQUESTED** [review]
hate to be that guy but just some formatting things.

File: app/config.py [hunk 3]
```diff
@@ -38,6 +40,8 @@ def int_or_none(key: str) -> Optional[int]:
             int(c) for c in env.pop("AUTOPUBLISH_USERS").split(" ")
         ]
 
+        self.join_messages = open("assets/join_messages").read().strip().split("\n")
```
Comment (trag1c) [inline app/config.py]: Change `.split("\n")` to `.splitlines()`, [here's why.](https://stackoverflow.com/a/22233816)

**Review by trag1c — CHANGES_REQUESTED** [review]

**Review by trag1c — CHANGES_REQUESTED** [review]
@Endercheif suggestions at `bot.py:179` & `bot.py:192`

**Review by trag1c — APPROVED** [review]

**Pull request merged**